### PR TITLE
Refactor/multipart stream

### PR DESCRIPTION
**Pull request by iilyak**

### CODE REVIEW

**Comment by kxepal** [comment]
Update: going to test this thoroughly during the week. I really miss most part of couch_httpd_multipart now ):

**Comment by iilyak** [comment]
@kxepal: Did you have chance to test it?

**Comment by kxepal** [comment]
@iilyak partially. So far it's all the fine. I'll give you more feedback over the May's day holidays.

File: src/couch_db_updater.erl [hunk 1]
```diff
@@ -605,10 +605,41 @@ init_db(DbName, Filepath, Fd, Header0, Options) ->
             Db
     end.
 
+open_db_file(Filepath, Options) ->
```
Comment (kxepal) [inline src/couch_db_updater.erl]: @iilyak It seems there are some commits that are not related to multipart refactor. Could you rebase your PR against current master?

Comment (iilyak) [inline src/couch_db_updater.erl]: I'll remove unrelated commits and rebase against current master.

Comment (kxepal) [inline src/couch_db_updater.erl]: Thanks! (:

**Comment by iilyak** [comment]
All unrelated commits are removed. PR is rebased against current master.
